src/lint/collect-algorithm-diagnostics.ts

```diff
@@ -33,13 +33,13 @@ function composeObservers(...observers: Observer[]): Observer {
 
 export function collectAlgorithmDiagnostics(
   report: (e: Warning) => void,
-  dom: any,
+  spec: Spec,
   sourceText: string,
   algorithms: { element: Element; tree?: EcmarkdownNode }[]
 ) {
   for (let algorithm of algorithms) {
     let element = algorithm.element;
-    let location = getLocation(dom, element);
+    let { source, ...location } = spec.locate(element);
```

Comment:
Having both `source` and `sourceText` in scope at the same time is a bit confusing. Consider renaming this?

Reply:
Renamed to `importSource` and `mainSource` respectively.